src/System.Net.Security/src/System/Net/Security/SslStream.cs

```diff
@@ -141,6 +145,17 @@ private X509Certificate UserCertSelectionCallbackWrapper(string targetHost, X509
             return _userCertificateSelectionCallback(this, targetHost, localCertificates, remoteCertificate, acceptableIssuers);
         }
 
+        private X509Certificate ServerCertSelectionCallbackWrapper(string targetHost)
+        {
+            return _userServerCertificateSelectionCallback(this, targetHost);
+        }
+
+        private void SetServerCertificateSelectionCallbackWrapper(SslServerAuthenticationOptions sslServerAuthenticationOptions)
+        {
+            _userServerCertificateSelectionCallback = sslServerAuthenticationOptions.ServerCertificateSelectionCallback;
+            sslServerAuthenticationOptions._serverCertDelegate = _userServerCertificateSelectionCallback == null ? null : new ServerCertCallback(ServerCertSelectionCallbackWrapper);
```

Comment:
Why do we need to create a new delegate?

Reply:
We made a decision in the past that each callback takes an `object sender` argument and because of that we need to wrap each delegate to pass SslStream - the delegate is used somewhere further down the stack which doesn't know about SslStream (we use exactly the same pattern for other callbacks)

Reply:
Is this mutating the SslServerAuthenticationOptions object supplied to the SslStream?

Reply:
I've created an issue to track this: #28657 since we already do it in other places I think I'd rather fix this outside of scope of this PR and fix all of the places